client/my-sites/site-settings/form-base.js

```diff
@@ -131,7 +139,9 @@ module.exports = {
 
 		notices.clearNotices( 'notices' );
 
-		this.setState( { submittingForm: true } );
+		//for dirtyFields, see lib/mixins/dirty-linked-state
+		this.setState( { submittingForm: true, dirtyFields: [] } );
+
 		site.saveSettings( this.state, function( error ) {
```

Comment:
I'll also add an `omit( this.state, 'dirtyFields' )` so we don't send over an unneeded field.